README.rst

```diff
@@ -24,16 +24,62 @@ Level, that is ``INFO`` by default, you could simply declare your configuration
 
 .. code-block:: python
 
-    class Configuration(ecological.AutoConfig):
+    class Configuration(ecological.Config):
         port: int
         debug: bool
         log_level: str = "INFO"
 
 And then set the environment variables ``PORT``, ``DEBUG`` and ``LOG_LEVEL``. ``Ecological`` will automatically set the
 class properties from the environment variables with the same (but upper cased) name.
 
-The values are set at the class definition type and assigned to the class itself (i.e. the class doesn't need to be
-instantiated).
+By default the values are set at the class definition type and assigned to the class itself (i.e. the class doesn't need to be
+instantiated). If needed this behavior can be changed (see the next section).
+
+Autoloading
+=============
+It is possible to defer/disable autoloading (setting) of variable values by specifyfing ``autoload`` option on class definition.
+
+On class creation (default)
+---------------------------
+When no option is provided values are loaded immediately on class creation and assigned to class attributes:
+
+.. code-block:: python
+
+    class Configuration(ecological.Config, autoload=ecological.Autoload.CLASS):
```

Comment:
If it's the default, it would perhaps be clearer to give a first example without the option at all? Basically for illustrative purposes.